apps/lib/http_server.c

```diff
@@ -294,23 +313,32 @@ int http_server_get_asn1_req(const ASN1_ITEM *it, ASN1_VALUE **preq,
 
     /* Read the request line. */
     len = BIO_gets(cbio, reqbuf, sizeof(reqbuf));
-    if (len <= 0) {
-        log_message(prog, LOG_INFO,
-                    "Request line read error or empty request");
+    if (len == 0)
+        return ret;
+    ret = 1;
+    if (len < 0) {
+        log_message(prog, LOG_WARNING, "Request line read error");
         (void)http_server_send_status(cbio, 400, "Bad Request");
         goto out;
     }
+    if ((end = strchr(reqbuf, '\r')) != NULL
+            || (end = strchr(reqbuf, '\n')) != NULL)
+        *end = '\0';
+    log_message(prog, LOG_INFO, "Received request, 1st line: %s", reqbuf);
 
     meth = reqbuf;
     url = meth + 3;
     if ((accept_get && strncmp(meth, "GET ", 4) == 0)
             || (url++, strncmp(meth, "POST ", 5) == 0)) {
+        const char *const http_version_str = " "HTTP_PREFIX_VERSION;
```

Comment:
Previously suggested in another PR:
```
        static const char http_version_str[] = " "HTTP_PREFIX_VERSION;
```

Reply:
Done - just cannot use `static` due to the subsequent line:
```
size_t http_version_str_len = strlen(http_version_str)
```
which the compiler cannot handle statically (even if I declare `http_version_str_len` static, too).

Reply:
You could use `http_version_str_len = sizeof(http_version_str) - 1;` if http_version_str is array.

Reply:
Good thought - done.